Tests/js/views/services/plugins/assets/ez-locationswapplugin-tests.js

```diff
@@ -0,0 +1,623 @@
+            });
+
+            this.service.fire('swapLocation', {});
+
+            Assert.isTrue(
+                contentDiscoverTriggered,
+                "The `contentDiscover` event should have been fired"
+            );
+        }
+    });
+
+    swapLocationTest = new Y.Test.Case({
+        name: "eZ Location Swap Plugin event tests",
+
+        setUp: function () {
+
+            this.service = new Y.Base();
+            this.view = new Y.View();
+            this.app = new Mock();
+            this.view.addTarget(this.service);
+            this.capi = new Mock();
+            this.contentServiceMock = new Mock();
+            this.contentJson = {
+                'id': '/content/Sergio/Aguero',
+                'name': 'Sergio Aguero',
+                'languageCode': 'esl-ES'
+            };
+            this.contentInfoMock = this._getContentInfoMock(this.contentJson);
+            this.locationJson = {
+                'id': '/location/Sergio/Aguero',
+                'contentInfo': this.contentInfoMock
+            };
+            this.location = this._getLocationMock(this.locationJson);
+            this.nonContainerContentTypeJson = {
+                'id': 'contentType',
+                'isContainer': false
+            };
+            this.nonContainerType = tests._getContentTypeMock(this.nonContainerContentTypeJson);
+
+            this.service.set('capi', this.capi);
+            this.service.set('location', this.location);
+            this.service.set('contentType', this.nonContainerType);
+
+            Mock.expect(this.capi, {
+                'method': 'getContentService',
+                'returns': this.contentServiceMock
+            });
+
+            this.plugin = new Y.eZ.Plugin.LocationSwap({
+                host: this.service
+            });
+        },
+
+        tearDown: function () {
+            this.plugin.destroy();
+            this.view.destroy();
+            this.service.destroy();
+            delete this.plugin;
+            delete this.view;
+            delete this.service;
+            delete this.app;
+        },
+
+        _getLocationMock: function (attrs) {
+            var locationMock = new Mock();
+
+            Mock.expect(locationMock, {
+                'method': 'get',
+                'args': [Mock.Value.String],
+                'run': function (attr) {
+                    switch (attr) {
+                        case 'id':
+                            return attrs.id;
+                        case 'contentInfo':
+                            return attrs.contentInfo;
+                        default:
+                            Assert.fail('Trying to `get` incorrect attribute');
+                            break;
+                    }
+                }
+            });
+
+            return locationMock;
+        },
+
+        _getContentInfoMock: function (attrs) {
+            var contentInfoMock = new Mock();
+
+            Mock.expect(contentInfoMock, {
```

Comment:
this very same code structure is repeated all over the test. You should define a function like the following to define a mock with a `get` method:
```js
function getModelMock(attributes) {
    var mock = new Mock();

    Mock.expect(mock, {
        method: 'get',
        args: [Mock.Value.String],
        run: function (attr) {
             if ( attributes[attr] ) {
                 return attributes[attr];
             } else {
                 Assert.fail("Unexpected call to get('" + attr + "')");
             }
        },
    });
    return mock;
}
```

Then for instance here, the method would become:
```js
_getContentInfoMock: function () {
    return getModelMock(attrs);
}
```